tests/common/util.cpp

```diff
@@ -0,0 +1,184 @@
+
+	EXPECT_EQ(NEXTPOWER2((uint32)  0),  1);
+
+	EXPECT_EQ(NEXTPOWER2((uint32)  2),  2);
+	EXPECT_EQ(NEXTPOWER2((uint32)  4),  4);
+	EXPECT_EQ(NEXTPOWER2((uint32)  8),  8);
+	EXPECT_EQ(NEXTPOWER2((uint32) 16), 16);
+}
+
+GTEST_TEST(Util, ARRAYSIZE) {
+	static const uint32 kArray[5] = { 0 };
+
+	EXPECT_EQ(ARRAYSIZE(kArray), 5);
+}
+
+GTEST_TEST(Util, convertIEEEFloatToFloat) {
+	EXPECT_FLOAT_EQ(convertIEEEFloat((uint32) 0x00000000),   0.00f);
+	EXPECT_FLOAT_EQ(convertIEEEFloat((uint32) 0x3F800000),   1.00f);
+	EXPECT_FLOAT_EQ(convertIEEEFloat((uint32) 0xBF800000), - 1.00f);
+	EXPECT_FLOAT_EQ(convertIEEEFloat((uint32) 0x41BC0000),  23.50f);
+	EXPECT_FLOAT_EQ(convertIEEEFloat((uint32) 0x40A75C29),   5.23f);
+}
+
+GTEST_TEST(Util, convertIEEEDoubleToDouble) {
+	EXPECT_DOUBLE_EQ(convertIEEEDouble(UINT64_C(0x0000000000000000)),   0.00);
+	EXPECT_DOUBLE_EQ(convertIEEEDouble(UINT64_C(0x3FF0000000000000)),   1.00);
+	EXPECT_DOUBLE_EQ(convertIEEEDouble(UINT64_C(0xBFF0000000000000)), - 1.00);
+	EXPECT_DOUBLE_EQ(convertIEEEDouble(UINT64_C(0x4037800000000000)),  23.50);
+	EXPECT_DOUBLE_EQ(convertIEEEDouble(UINT64_C(0x4014EB851EB851EC)),   5.23);
+}
+
+GTEST_TEST(Util, convertIEEEFloatFromFloat) {
+	EXPECT_EQ(convertIEEEFloat(  0.00f), (uint32) 0x00000000);
+	EXPECT_EQ(convertIEEEFloat(  1.00f), (uint32) 0x3F800000);
+	EXPECT_EQ(convertIEEEFloat(- 1.00f), (uint32) 0xBF800000);
+	EXPECT_EQ(convertIEEEFloat( 23.50f), (uint32) 0x41BC0000);
+	EXPECT_EQ(convertIEEEFloat(  5.23f), (uint32) 0x40A75C29);
+}
+
+GTEST_TEST(Util, convertIEEEDoubleFromDouble) {
+	EXPECT_EQ(convertIEEEDouble(  0.00), UINT64_C(0x0000000000000000));
+	EXPECT_EQ(convertIEEEDouble(  1.00), UINT64_C(0x3FF0000000000000));
+	EXPECT_EQ(convertIEEEDouble(- 1.00), UINT64_C(0xBFF0000000000000));
+	EXPECT_EQ(convertIEEEDouble( 23.50), UINT64_C(0x4037800000000000));
+	EXPECT_EQ(convertIEEEDouble(  5.23), UINT64_C(0x4014EB851EB851EC));
+}
+
+GTEST_TEST(Util, readIEEEFloat16) {
+	EXPECT_FLOAT_EQ(readIEEEFloat16(0x0000),   0.00f);
+	EXPECT_FLOAT_EQ(readIEEEFloat16(0x3C00),   1.00f);
+	EXPECT_FLOAT_EQ(readIEEEFloat16(0xBC00), - 1.00f);
+	EXPECT_FLOAT_EQ(readIEEEFloat16(0x4DE0),  23.50f);
+
+	// 16-bit half-precision floats can't represent 5.23 more accurately
+	EXPECT_NEAR(readIEEEFloat16(0x453B), 5.23f, 0.0005);
+}
+
+GTEST_TEST(Util, readNintendoFixedPoint) {
+	EXPECT_DOUBLE_EQ(readNintendoFixedPoint(0x00000000,  true, 15, 16),      0.00);
+	EXPECT_DOUBLE_EQ(readNintendoFixedPoint(0x00010000,  true, 15, 16),      1.00);
+	EXPECT_DOUBLE_EQ(readNintendoFixedPoint(0xFFFF0000,  true, 15, 16), -    1.00);
+	EXPECT_DOUBLE_EQ(readNintendoFixedPoint(0xFFFF0000, false, 16, 16),  65535.00);
+
+	/* The fractional part in Nintendo's fixed point format is expressed in simple
+	 * fractions of 2^fBits. That's pretty different from IEEE floats, so their
+	 * representation of non-trivial fractions diverge quite a bit. Which means
+	 * checking that the result is within a few double ULP doesn't make much sense.
+	 */
```

Comment:
(I already talked with @clone2727 about this, so this is just for the record: nope, they have to be treated as floating point numbers, because the numbers are thrown into Matrix4x4 and OpenGL.)